azure-iot-device/azure/iot/device/common/auth/sastoken.py

```diff
@@ -0,0 +1,96 @@
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for
+# license information.
+# --------------------------------------------------------------------------
+"""This module contains tools for working with Shared Access Signature (SAS) Tokens"""
+
+import base64
+import hmac
+import hashlib
+import time
+import six.moves.urllib as urllib
+from azure.iot.device.common.chainable_exception import ChainableException
+
+
+class SasTokenError(ChainableException):
+    """Error in SasToken"""
+
+    pass
+
+
+class SasToken(object):
+    """Shared Access Signature Token used to authenticate a request
+
+    Data Attributes:
+    expiry_time (int): Time that token will expire (in UTC, since epoch)
+    ttl (int): Time to live for the token, in seconds
+    """
+
+    _auth_rule_token_format = (
```

Comment:
>_auth_rule_token_format [](start = 4, length = 23)

why not  `keyname_token_format` ?

Reply:
Believe it or not.... according to official docs, "auth rule" is the official name. I was also kind of shocked.

Reply:
Because apparently for some reason the official name of of the `skn` field is not... `shared key name` as you might expect, and is for some reason... `auth rule`.

I don't get it either.

---
In reply to: [426946266](https://github.com/Azure/azure-iot-sdk-python/pull/555#discussion_r426946266) [](ancestors = 426946266)

Reply:
I'm wondering if this was the name in the azure portal at one point.  Right now, the key names (iothubowner, etc) show up in the portal under "Shared Access Policy" as "Access policy name".  We had this same thing happen with one of the DPS names -- I think ProvisioningHostName shows up in the portal with one name and in the SDK as another.  

---
In reply to: [427500301](https://github.com/Azure/azure-iot-sdk-python/pull/555#discussion_r427500301) [](ancestors = 427500301,426946266)